flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/scalar/JsonUnquoteFunction.java

```diff
@@ -0,0 +1,128 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.functions.scalar;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.data.binary.BinaryStringData;
+import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
+import org.apache.flink.table.functions.SpecializedFunction.SpecializedContext;
+import org.apache.flink.table.runtime.functions.SqlJsonUtils;
+
+import javax.annotation.Nullable;
+
+/** Implementation of {@link BuiltInFunctionDefinitions#JSON_UNQUOTE}. */
+@Internal
+public class JsonUnquoteFunction extends BuiltInScalarFunction {
+
+    public JsonUnquoteFunction(SpecializedContext context) {
+        super(BuiltInFunctionDefinitions.JSON_UNQUOTE, context);
+    }
+
+    private static boolean isValidJsonVal(String jsonInString) {
+        // See also BuiltInMethods.scala, IS_JSON_VALUE
+        return SqlJsonUtils.isJsonValue(jsonInString);
+    }
+
+    private String unescapeStr(String inputStr) {
+        StringBuilder result = new StringBuilder();
+        int i = 0;
+        while (i < inputStr.length()) {
+            if (inputStr.charAt(i) == '\\' && i + 1 < inputStr.length()) {
+                i++; // move to the next char
+                char ch = inputStr.charAt(i++);
+
+                switch (ch) {
+                    case '"':
+                        result.append(ch);
+                        break;
+                    case '\\':
+                        result.append(ch);
+                        break;
+                    case '/':
+                        result.append(ch);
+                        break;
+                    case 'b':
+                        result.append('\b');
+                        break;
+                    case 'f':
+                        result.append('\f');
+                        break;
+                    case 'n':
+                        result.append('\n');
+                        break;
+                    case 'r':
+                        result.append('\r');
+                        break;
+                    case 't':
+                        result.append('\t');
+                        break;
+                    case 'u':
+                        result.append(fromUnicodeLiteral(inputStr, i));
+                        i = i + 4;
+                        break;
+                    default:
+                        throw new RuntimeException("Illegal escape sequence: \\" + ch);
```

Comment:
```suggestion
                        throw new IllegalArgumentException("Illegal escape sequence: \\" + ch);
```

Reply:
IMHO: since it is already runtime shouldn't it be `FlinkRuntimeException` here?
https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/util/FlinkRuntimeException.java

which is already used within builtin functions at https://github.com/apache/flink/tree/master/flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/scalar